driver/modern_bpf/helpers/extract/extract_from_kernel.h

```diff
@@ -106,15 +156,25 @@ static __always_inline unsigned long extract__syscall_argument(struct pt_regs *r
  */
 static __always_inline void extract__network_args(void *argv, int num, struct pt_regs *regs)
 {
-#ifdef CAPTURE_SOCKETCALL
 	int id = extract__syscall_id(regs);
+	bool is_32bit_syscall = extract__32bit_syscall();
+#ifdef __NR_socketcall
 	if(id == __NR_socketcall)
+#elif defined(__TARGET_ARCH_x86)
+	if(is_32bit_syscall && id == __NR_ia32_socketcall)
+#else
+	if (false)
+#endif
 	{
+		size_t size = sizeof(unsigned long);
+		if (is_32bit_syscall)
+		{
+			size = sizeof(u32);
+		}
 		unsigned long args_pointer = extract__syscall_argument(regs, 1);
-		bpf_probe_read_user(argv, num * sizeof(unsigned long), (void*)args_pointer);
+		bpf_probe_read_user(argv, num * size, (void*)args_pointer);
```

Comment:
This change is breaking the verifier (not on my machine :D ).
I think the problem here is that we are receiving an `unsigned long` arg from the ia32 user program (that is 4B wide), but all our logic uses `unsigned long` that are 8B wide.
I will think about a smarter way to handle all of this :/ 
If you just read 3 * 8B (24B),  we will read a correct first param, first 4B of second param and completely random data for third param (because user's pushed data is 12B long).

The same change had to be applied to old bpf probe too, see `bpf_syscall_get_socketcall_arg`.

Reply:
Fixed by using a slightly different code.